### PR TITLE
[release/v2.9] Add functionality to set condition on v3 Cluster to indicate CA validity

### PR DESCRIPTION
**Pull request by Oats87**
## Issue: 
https://github.com/rancher/rancher/issues/45581
https://github.com/rancher/rancher/issues/45582
 
## Problem
In certain cases, it is possible that the `rancher-agent` and `system-agent` connect to Rancher without properly validating the cacerts.

## Solution
This PR introduces logic that enables Rancher to operate using a setting (`agent-tls-mode`). When set to `strict`, the `rancher-agent` and `cattle-cluster-agent` will both retrieve and utilize the certificates served by rancher at the `/cacerts` path, and only trust that certificate chain for connection to Rancher.

### Manual Testing
I tested a few cases:
1. Rancher has `--no-cacerts` and `agent-tls-mode=strict` -- as expected, nothing works
2. Rancher has a valid `cacerts` chain and `agent-tls-mode=strict` -- everything works
3. Rancher has an invalid `cacerts` chain and `agent-tls-mode=strict` -- nothing works
4. Rancher has an invalid `cacerts` chain and `agent-tls-mode=system-store` - everything works but the condition is set on the v3 clusters that says it is not a valid chain

### Automated Testing

* Test types added/modified:
    * Unit
Summary: Added unit tests for the new `cavalidator` controller

### Regressions Considerations

Existing / newly added automated tests that provide evidence there are no regressions:

### CODE REVIEW

File: pkg/agent/cluster/cluster.go [hunk 1]
```diff
@@ -49,6 +49,10 @@ func CAChecksum() string {
 	return os.Getenv("CATTLE_CA_CHECKSUM")
 }
 
+func CAStrictVerify() bool {
+	return strings.ToLower(os.Getenv("STRICT_VERIFY")) == "true"
```
Comment (kinarashah) [inline pkg/agent/cluster/cluster.go:53]: Would it make more sense to name this var `CATTLE_CA_STRICT_VERIFY` in line with our agent env vars?

File: pkg/controllers/managementuser/cavalidator/cavalidator.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+				}
+			}
+			return obj, nil
+		}
+		newMgmtCluster := mgmtCluster.DeepCopy()
+		if string(obj.Data[CacertsValid]) == "false" {
+			CertificateAuthorityValid.False(newMgmtCluster)
+		} else {
+			CertificateAuthorityValid.Unknown(newMgmtCluster)
+		}
```
Comment (kinarashah) [inline pkg/controllers/managementuser/cavalidator/cavalidator.go:63]: Unless we add `needsUpdate()` check here, we'd be doing deepCopy + update unnecessarily...not sure if it's expensive enough for consideration.

File: pkg/capr/installer/installer.go [hunk 3]
```diff
@@ -80,6 +80,28 @@ func LinuxInstallScript(ctx context.Context, token string, envVars []corev1.EnvV
 	if token != "" {
 		token = "CATTLE_ROLE_NONE=true\nCATTLE_TOKEN=\"" + token + "\""
 	}
+
+	// Merge the env vars with the AgentTLSModeStrict
+	found := false
+	for _, ev := range envVars {
+		if ev.Name == "STRICT_VERIFY" {
```
Comment (jakefhyde) [inline pkg/capr/installer/installer.go:87]: `"STRICT_VERIFY"` could probably be a constant in `capr/common.go`.

Comment (Oats87) [reply pkg/capr/installer/installer.go:87]: I didn't want to make it a constant on our side yet given it's used for the `cluster-agent` as well as `system-agent`. We can revisit this later though

File: pkg/agent/cluster/cluster.go (hunk 1 above)
Comment (Oats87) [reply pkg/agent/cluster/cluster.go:53]: This is actually being called `STRICT_VERIFY` as it is being driven by the design doc. I've copied the environment variable name verbatim out of the doc.

File: pkg/controllers/managementuser/cavalidator/cavalidator.go (hunk 2 above)
Comment (Oats87) [reply pkg/controllers/managementuser/cavalidator/cavalidator.go]: I added functionality to determine if we need to call `DeepCopy` and `Update`

**Review by kinarashah — APPROVED** [review]

**Review by jakefhyde — APPROVED** [review]

**Pull request merged**